src/v/config/configuration.cc

```diff
@@ -724,6 +724,12 @@ configuration::configuration()
       "to simplify testing and shouldn't be set in production.",
       {.needs_restart = needs_restart::yes, .visibility = visibility::tunable},
       false)
+  , log_compaction_use_sliding_window(
+      *this,
+      "log_compaction_use_sliding_window",
+      "Use sliding window compaction.",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      false)
```

Comment:
do you think this should require a restart? looking at the few places where it is checked, it's not obvious what kind assumptions there are for example between storage and archival where they seem to coordinate on their view of the type of compaction currently enabled which might not be well coordinated if the config changes interleaves in certain ways with execution? maybe its safe tho.

Reply:
Yeah this is a good point. It'll be less error prone to require a restart. Will update

Reply:
Done